src/utils/internal.ts

```diff
@@ -534,3 +535,17 @@ export function assertFormatValid(address: string, ss58Format: number): void {
     });
   }
 }
+
+/**
+ * @hidden
+ */
+export function getTicker(token: string | SecurityToken): string {
+  return typeof token === 'string' ? token : token.ticker;
+}
+
+/**
+ * @hidden
+ */
+export function calculateMoment(date: Date | undefined, context: Context): Moment | null {
+  return (date && dateToMoment(date, context)) ?? null;
+}
```

Comment:
We can generalize this to be able to use it with different conversion functions and types

```
/**
 * @hidden
 * Transform a conversion util into a version that returns null if the input is falsy
 */ 
export function optionize<InputType, OutputType>(converter: (input: InputType, context: Context) => OutputType): (val: InputType | null | undefined, context: Context) => OutputType | null {
  return (value: InputType | null = null, context: Context): OutputType | null => value && converter(value, context);
}
```

It receives a conversion function and returns a version of the function that returns null if the argument is falsy

Example:

```
const optionizedFunc = optionize(dateToMoment);
const maybeNull = optionizedFunc(someDate, context); // Moment | null

// or even
const result = optionize(dateToMoment)(someDate, context); // Moment | null
```